src/notation/internal/notationinteraction.cpp

```diff
@@ -2094,56 +2094,52 @@ bool NotationInteraction::isGripEditStarted() const
     return m_gripEditData.element && m_gripEditData.curGrip != Ms::Grip::NO_GRIP;
 }
 
-bool NotationInteraction::isHitGrip(const PointF& pos) const
+static int findGrip(const QVector<mu::RectF>& grips, Ms::Page* page, const mu::PointF& canvasPos)
 {
-    if (!selection()->element() || m_gripEditData.grip.empty()) {
-        return false;
+    if (grips.empty() || page == nullptr) {
+        return -1;
     }
-
-    qreal align = m_gripEditData.grip[0].width() / 2;
-
-    for (int i = 0; i < m_gripEditData.grips; ++i) {
-        if (m_gripEditData.grip[i].adjusted(-align, -align, align, align).contains(pos)) {
-            return true;
+    mu::PointF pos = canvasPos - page->pos();
+    qreal align = grips[0].width() / 2;
+    for (int i = 0; i < grips.size(); ++i) {
+        if (grips[i].adjusted(-align, -align, align, align).contains(pos)) {
+            return i;
         }
     }
+    return -1;
+}
 
-    return false;
+bool NotationInteraction::isHitGrip(const PointF& pos) const
+{
+    return selection()->element() && findGrip(m_gripEditData.grip, point2page(pos), pos) != -1;
```

Comment:
You don't use selection()->element() here, so you can remove it from here

Reply:
`isHitGrip` should return `false` if  `selection()->element()` is `nullptr`.
In principle `m_gripEditData.grip` should always be an empty array in that case, but the check was there already.